src/components/Header.tsx

```diff
@@ -0,0 +1,295 @@
+export interface HeaderProps {}
+
+const Header: React.FC<HeaderProps> = () => {
+  const { activate, account, deactivate } = useWeb3React<Web3Provider>();
+
+  const showLoginDialog = useHeaderStore((s) => s.showDialog);
+  const setShowLoginDialog = useHeaderStore((s) => s.setShowDialog);
+  const setHasTriedEagerConnecting = useHeaderStore((s) => s.setHasTriedEagerConnecting);
+
+  const userEnsName = useEns();
+
+  const hasTriedEagerConnect = useEagerConnect();
+
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  useEffect(() => {
+    setHasTriedEagerConnecting(hasTriedEagerConnect);
+  }, [hasTriedEagerConnect, setHasTriedEagerConnecting]);
+
+  const [currentlyActivating, setCurrentlyActivating] = useState<'metamask' | 'wc' | undefined>(undefined);
+
+  const activeWithMetamask = useCallback(async () => {
+    setErrorMessage(null);
+    setCurrentlyActivating('metamask');
+    try {
+      await activate(injected, undefined, true);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (e: any) {
+      console.log(e.name);
+      if (e.name === 'NoEthereumProviderError' || e.message?.includes('No Ethereum provider was found')) {
+        setErrorMessage('No MetaMask detected.');
+      } else if (
+        e.message.includes('Already processing eth_requestAccounts') ||
+        e.message.includes(`Request of type 'wallet_requestPermissions'`)
+      ) {
+        setErrorMessage('Check MetaMask for an existing login request');
+      } else if (e.message.includes('The user rejected the request')) {
+        setErrorMessage('The MetaMask login was closed, try connecting again');
+      } else {
+        setErrorMessage(e.message ?? 'Something went wrong logging in');
+        console.log(e, Object.keys(e));
+      }
+      setCurrentlyActivating(undefined);
+      return;
+    }
+    setShowLoginDialog(false);
+    setTimeout(() => {
+      setCurrentlyActivating(undefined);
+    }, 1500);
+  }, [activate, setShowLoginDialog]);
+
+  const activeWithWalletConnect = useCallback(async () => {
+    setErrorMessage(null);
+    setCurrentlyActivating('wc');
+    try {
+      await activate(walletconnect, undefined, true);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (e: any) {
+      setCurrentlyActivating(undefined);
+      setErrorMessage(e.message ?? 'Something went wrong logging in with WalletConnect');
+      console.log(e);
+      return;
+    }
+    setShowLoginDialog(false);
+    setTimeout(() => {
+      setCurrentlyActivating(undefined);
+    }, 1500);
+  }, [activate, setShowLoginDialog]);
+
+  const logoutUser = useCallback(async () => {
+    deactivate();
+  }, [deactivate]);
+
+  useInactiveListener(!hasTriedEagerConnect);
+
+  return (
+    <>
+      <HeaderWrapperOuter>
+        <HeaderWrapperInner>
+          <LeftWrapper>
+            <Link to={'/'}>
+              <Logo />
```

Comment:
I think you are getting console errors from nesting a link tag like this

Warning: validateDOMNesting(...): <a> cannot appear as a descendant of <a>.

Reply:
nice page though

Reply:
I've fixed this up, I think we can merge now.